lib/src/components/NxAlert/__tests__/NxAlert.test.tsx

```diff
@@ -4,263 +4,158 @@
+        const view = quickRender(),
+            viewWithRole = quickRender({ role: 'foo' });
 
-      it('renders the children passed into it', function() {
-        const nxWarningAlert = getNxWarningAlert();
-        expect(nxWarningAlert).toContainReact(<p>Hello</p>);
+        expect(view.container.firstElementChild).not.toHaveAttribute('role');
+        expect(viewWithRole.container.firstElementChild).toHaveAttribute('role', 'foo');
       });
 
-      it('passes any other props to the NxAlert', function() {
-        const onClose = jest.fn(),
-            component = getNxWarningAlert({ id: 'foo', title: 'baz', onClose });
-        expect(component).toHaveProp('id', 'foo');
-        expect(component).toHaveProp('title', 'baz');
-        expect(component).toHaveProp('onClose', onClose);
+      it('renders an img with an accessible nameof "Info"', function() {
+        expect(quickRender().getByRole('img')).toHaveAccessibleName('Info');
       });
     });
 
-    describe('NxInfoAlert', function() {
-      const getNxInfoAlert = enzymeUtils.getShallowComponent<Props>(NxInfoAlert, minimalProps);
-
-      it('renders an NxAlert', function() {
-        const nxInfoAlert = getNxInfoAlert();
-        expect(nxInfoAlert).toMatchSelector(NxAlert);
-      });
-
-      it('sets the iconLabel to "Info"', function() {
-        expect(getNxInfoAlert()).toHaveProp('iconLabel', 'Info');
-      });
+    describe('NxWarningAlert', function() {
+      baseTests(NxWarningAlert, minimalProps);
 
-      it('renders the appropriate info classes', function() {
-        const nxInfoAlert = getNxInfoAlert();
-        expect(nxInfoAlert).toMatchSelector(NxAlert);
-        expect(nxInfoAlert).toHaveClassName('nx-alert--info');
-      });
+      const quickRender = rtlRender(NxWarningAlert, minimalProps);
 
-      it('renders the appropriate info icon', function() {
-        const nxInfoAlert = getNxInfoAlert();
-        expect(nxInfoAlert).toMatchSelector(NxAlert);
-        expect(nxInfoAlert).toHaveProp('icon', faInfoCircle);
-      });
+      it('sets the top-level element\'s role from the props', function() {
+        const view = quickRender(),
+            viewWithRole = quickRender({ role: 'foo' });
 
-      it('renders the children passed into it', function() {
-        const nxInfoAlert = getNxInfoAlert();
-        expect(nxInfoAlert).toContainReact(<p>Hello</p>);
+        expect(view.container.firstElementChild).not.toHaveAttribute('role');
+        expect(viewWithRole.container.firstElementChild).toHaveAttribute('role', 'foo');
       });
 
-      it('passes any other props to the NxAlert', function() {
-        const onClose = jest.fn(),
-            component = getNxInfoAlert({ id: 'foo', title: 'baz', onClose });
-        expect(component).toHaveProp('id', 'foo');
-        expect(component).toHaveProp('title', 'baz');
-        expect(component).toHaveProp('onClose', onClose);
+      it('renders an img with an accessible nameof "Warning"', function() {
```

Comment:
```suggestion
      it('renders an img with an accessible name of "Warning"', function() {
```

Reply:
(in all `it('renders an img with an accessible nameof...)`

Reply:
3c5b1c0a2